image/src/app/module/main.py

```diff
@@ -15,18 +16,30 @@ def module_main():
     Returns:
         [string, string]: [data, error]
     """
+    parity_dict = {
+       "None" : serial.PARITY_NONE,
+       "Even" : serial.PARITY_EVEN,
+       "Odd"  : serial.PARITY_ODD
+    }
+    if APPLICATION['PARITY'] not in parity_dict:
+        raise Exception("Invalid parity")
     try:
         #  open the serial port and get the serial port object
-        ser = serial.Serial(APPLICATION['PORT'], APPLICATION['BAUD_RATE'], timeout=1,bytesize=APPLICATION['DATA_BITS'],parity=APPLICATION['PARITY'], stopbits=APPLICATION['STOP_BITS'])
-        while True: 
-          #  read json payload
-          ser.reset_input_buffer()
-          data = ser.readline().decode("ISO-8859-1")
-          dict_json = json.loads(data)
-          sent=send_data(dict_json)
-          if sent:
-            print("SUCCESS"),print("failed")
-          print(dict_json)
-    except json.JSONDecodeError as e:
-      return None, f"Unable to perform the module logic: {e}"
-      
+        str_data_bytes = bytes(str(APPLICATION['DATA_BITS']), "ascii")
```

Comment:
the intermediary step with bytes seems unnecessary

Reply:
I didn't update the code yet, but yes it is unnecessary , just thinking and coding without removing the unessassary